File: AutoSchedulerBE/AutoScheduler.Domain/Entities/MemberGroups/Group.cs
using AutoScheduler.Domain.Entities.Activities;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace AutoScheduler.Domain.Entities.MemberGroups
{
    public class Group
    {
        public int Id { get; set; }
        public int OrganizationId { get; set; }
        public required string Name { get; set; }
        public string? Description { get; set; }
        public int? ParentGroupId { get; set; }
        public IList<Group>? SubGroups { get; set; }
        public IList<ActivityRequirements>? Requirements { get; set; }

    }
}

File: AutoSchedulerFE/src/classes/group.ts
import type { Activity, ActivityRequirements, Hall } from "./activity"

export interface Group{
    id: number,
    organizationId: number,
    name: string,
    description: string | undefined,
    parentGroupId: number | undefined
    subGroups: Group[],
    requirements: ActivityRequirements[]
};

export interface Member{
    id: number,
    organizationId : number | undefined,
    name: string,
    contact: string | undefined,
    availability: Availability[]
};

export interface Organization{
    id: number,
    name: string,
    description: string | undefined,
    groups: Group[],
    halls: Hall[],
    activities: Activity[],
    members: Member[]
};

export interface Availability{
    id: number,
    startTime: string,
    endTime: string,
    dayOfTheWeek: string
};